protoc_plugin/lib/message_generator.dart

```diff
@@ -523,6 +523,19 @@ class MessageGenerator extends ProtobufContainer {
                 fieldPathSegment: memberFieldPath,
                 start: 'void '.length)
           ]);
+      if (field.baseType.isMessage) {
+        out.printlnAnnotated(
+            '${fieldTypeString} ${names.ensureMethodName}() => \$_ensure('
+            '${field.index}, '
```

Comment:
Try to pass only the index.

You can create a new instance with FieldInfo.subBuilder